File: Medium/2026-05-27-3121-Count-the-Number-of-Special-Characters-II/solution.py
class Solution:
    def countSpecialCharacters(self, word: str) -> int:
        special_count = 0
        seen_lower = set()
        seen_upper = set()
        
        for char in word:
            if char.islower():
                seen_lower.add(char)
            elif char.isupper():
                seen_upper.add(char.lower())
        
        for char in seen_upper:
            if char in seen_lower:
                first_upper_index = word.index(char.upper())
                first_lower_index = word.index(char.lower())
                if first_lower_index < first_upper_index:
                    special_count += 1
        
        return special_count

File: Medium/2026-05-27-3121-Count-the-Number-of-Special-Characters-II/solution.java
class Solution {
    public int countSpecialCharacters(String word) {
        int count = 0;
        boolean[] seenLower = new boolean[26];
        boolean[] seenUpper = new boolean[26];
        
        for (char c : word.toCharArray()) {
            if (Character.isLowerCase(c)) {
                seenLower[c - 'a'] = true;
            } else {
                seenUpper[c - 'A'] = true;
            }
        }
        
        for (int i = 0; i < 26; i++) {
            if (seenLower[i] && seenUpper[i] && isLowerBeforeUpper(word, (char) (i + 'a'), (char) (i + 'A'))) {
                count++;
            }
        }
        
        return count;
    }
    
    private boolean isLowerBeforeUpper(String word, char lower, char upper) {
        boolean lowerFound = false;
        for (char c : word.toCharArray()) {
            if (c == lower) {
                lowerFound = true;
            } else if (c == upper) {
                return lowerFound;
            }
        }
        return false;
    }
}